cdap-ui/app/directives/epoch/charts.js

```diff
@@ -52,6 +56,9 @@ ngEpoch.controller('epochController', function ($scope, $compile, caskWindowMana
       $scope.$watch('history', function (newVal) {
         if(newVal) {
           $scope.options.data = newVal;
+          if ($scope.options.updateFull) {
```

Comment:
We could probably render the entire history again right? Why update an entire history and introduce a special flag for it?